pkg/cpio/cpio.go

```diff
@@ -81,6 +81,27 @@ func NewDedupWriter(rw RecordWriter) RecordWriter {
 	}
 }
 
+// Passthrough copies from a RecordReader to a RecordWriter
+// It processes one record at a time to minimize the memory footprint.
+func Passthrough(r RecordReader, w RecordWriter) error {
+	for {
+		rec, err := r.ReadRecord()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf("error reading records: %v", err)
+		}
+		if err := w.WriteRecord(rec); err != nil {
+			return fmt.Errorf("Writing record %q failed: %v", rec, err)
```

Comment:
this error does. 

I suggest you make the errors act the same way i.e. include the record in the error string